papas/aliceproto/test_reconstruction.py

```diff
@@ -0,0 +1,305 @@
+                print "made hadron" 
+                
+        elif  ((len(parents)==3)  & (block.countECAL()==1) & (block.countTrack()==1) & (block.countHCAL()==1)) :
+                print "made hadron and photon"
+                
+                #probably not right but illustrates splitting of parents
+                hparents=[] # will contain parents for the Hadron
+                for elem in parents:
+                    if (Identifier.isHCAL(elem)) :
+                        self.makePhoton({elem})
+                    else :
+                        hparents.append(elem)    
+                    
+                self.makeHadron(hparents)
+    
+        else :
+            print "particle TODO"  
+         
+    def makePhoton(self,parents) :
+        return self.addParticle(self.newID(), 22,parents)
+
+    def makeHadron(self,parents) :
+        return self.addParticle(self.newID(), 211,parents)
+
+    def addParticle(self, id,pdgid,parents) :
+        particle=recParticle(id,pdgid)
+        self.event.recParticles[particle.uniqueid]= particle
+        particleNode=Node(particle.uniqueid)
+        self.event.historyNodes[particle.uniqueid]= particleNode
+        for parent in parents :
+            self.event.historyNodes[parent].add_child(particleNode)
+        
+    def newID(self):
+        id=self.particlecount
+        self.particlecount+=1
+        return id
+
+class Cluster(object):
+    def __init__(self, id, layer):
+        if (layer=='ecal_in') :
+            self.uniqueid=Identifier.makeID(self,Identifier.PFOBJECTTYPE.ECALCLUSTER)
+        elif (layer=='hcal_in') :
+            self.uniqueid=Identifier.makeID(self,Identifier.PFOBJECTTYPE.HCALCLUSTER)
+        else:
+            assert false
+        self.layer=layer
+        self.id=id
+        
+class Track(object):
+    def __init__(self, id):
+        self.uniqueid=Identifier.makeID(self,Identifier.PFOBJECTTYPE.TRACK)
+        self.id=id
+        self.layer='tracker'
+
+class Particle(object):
+    def __init__(self, id,pdgid):
+        self.uniqueid=Identifier.makeID(self,Identifier.PFOBJECTTYPE.PARTICLE)
+        self.pdgid=pdgid
+        self.id=id
+        #self.type=PFType.PARTICLE
+        
+class recParticle(Particle):
+    def __init__(self, id,pdgid):
+        self.uniqueid=Identifier.makeID(self,Identifier.PFOBJECTTYPE.RECPARTICLE)
+        self.pdgid=pdgid
+        self.id=id
+        #print id
+        #self.type=PFType.RECPARTICLE
+
+
+
+#class PFType(Enum):
+    #NONE = 0
+    #TRACK = 1
+    #ECAL = 2
+    #HCAL = 4
+    #PARTICLE=8
+    #RECPARTICLE=16
+    
+#class PFDISTANCEtype(Enum):
+    #NONE = 0
+    #TRACK_TRACK = 2
+    #TRACK_ECAL = 3
+    #ECAL_ECAL = 4
+    #TRACK_HCAL=5
+    #ECAL_HCAL = 6
+    #HCAL_HCAL=8
+           
+    
+def DistanceItem(obj1,obj2): #simple distance uses the short ids
+    distance=abs(obj1.id%100 -obj2.id%100)
+    return  None, distance==0, distance
+         
+        
+class TestFloodFill(unittest.TestCase):
```

Comment:
looks like this is the only test case, so maybe the module should be named differently